cli/pkg/update/update.go

```diff
@@ -0,0 +1,209 @@
+package update
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/hashicorp/go-version"
+	"github.com/rilldata/rill/cli/pkg/dotrill"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	StateFilename       = "state.yaml"
+	VersionInfoStateKey = "version_info"
+	Addr                = "https://api.github.com/repos/rilldata/rill-developer/releases/latest"
+)
+
+type UpdateInfo struct {
+	Update      bool
+	Message     string
+	ReleaseInfo *ReleaseInfo
+}
+
+// ReleaseInfo stores information about a release
+type ReleaseInfo struct {
+	Version     string    `json:"tag_name"`
+	URL         string    `json:"html_url"`
+	PublishedAt time.Time `json:"published_at"`
+}
+
+func CheckVersion(ctx context.Context, currentVersion string) error {
+	// Check if build from source
+	if currentVersion == "" {
+		return nil
+	}
+
+	updateInfo, err := checkVersion(ctx, currentVersion)
+	if err != nil {
+		return err
+	}
+
+	if updateInfo.Update {
+		fmt.Printf("\n%s %s → %s\n\n",
+			color.YellowString("A new version of rill is available:"),
+			color.CyanString(currentVersion),
+			color.CyanString(updateInfo.ReleaseInfo.Version))
+	}
+
+	return nil
+}
+
+func checkVersion(ctx context.Context, currentVersion string) (*UpdateInfo, error) {
```

Comment:
Replace the version getting and checking logic here with a `LatestVersion` function that returns the cached version number or calls `fetchLatestVersion` if cache is outdated.

Reply:
It should just return the newest version number, nothing else

Reply:
Then move the version checking logic into `CheckVersion`